src/Processors/QueryPlan/Optimizations/filterPushDown.cpp

```diff
@@ -176,6 +176,9 @@ size_t tryPushDownFilter(QueryPlan::Node * parent_node, QueryPlan::Nodes & nodes
 
     if (auto * aggregating = typeid_cast<AggregatingStep *>(child.get()))
     {
+        if (aggregating->isGroupingSets())
+            return 0;
```

Comment:
In this case we can apply optimisation if all identifiers in the predicate are use in each grouping set